crates/astria-conductor/src/executor.rs

```diff
@@ -272,50 +313,25 @@ impl<C: ExecutionClient> Executor<C> {
                 // try executing the block as it hasn't been executed before
                 // execute_block will check if our namespace has txs; if so, it'll return the
                 // resulting execution block hash, otherwise None
-                let Some(execution_block_hash) = self
-                    .execute_block(block)
+                let Some(executed_block) = self
+                    .execute_block(block.clone())
                     .await
                     .wrap_err("failed to execute block")?
                 else {
                     // no txs for our namespace, nothing to do
-                    debug!("execute_block returned None; skipping finalize_block");
+                    debug!("execute_block returned None; skipping call_update_commitment_state");
                     return Ok(());
                 };
-
-                // finalize the block after it's been executed
-                self.finalize_block(execution_block_hash, sequencer_block_hash)
+                // when we execute a block received from da, nothing else has been executed on top
+                // of it, so we set FIRM and SOFT to this executed block
+                self.update_commitment_state(executed_block.clone(), executed_block)
```

Comment:
this looks weird - i also noticed we are usually only updating `soft` OR `firm`, not both at the same time (except for here), so i'd add functions `update_soft_commitment` and `update_firm_commitment` to make it clear which is being updated